fuse_core/include/fuse_core/throttled_callback.h

```diff
@@ -190,22 +280,40 @@ class ThrottledCallback
     callback(std::forward<Args>(args)...);
   }
 
-private:
-  Callback keep_callback_;         //!< The callback to call when kept, i.e. not dropped
-  Callback drop_callback_;         //!< The callback to call when dropped because of throttling
-  ros::Duration throttle_period_;  //!< The throttling period duration in seconds
-  bool use_wall_time_;             //<! The flag to indicate whether to use ros::WallTime or not
 
-  ros::Time last_called_time_;  //!< The last time the keep callback was called
+  /**
+   * @brief Callback that uses message header timestamps to throttle the calls to the keep callback provided.
+   * When dropped because of throttling, the drop callback is called instead.
+   * @details The first message after the most recent whole-number multiple of the throttle_period_ is published
+   * and all others are dropped.
+   *
+   * @param[in] msg The message to be throttled
+   */
+  template <typename = typename std::enable_if_t<ros::message_traits::HasHeader<M>::value>>
+  void messageStampCallback(const typename M::ConstPtr& msg)
+  {
+    const auto msg_stamp_time = ros::Time(msg->header.stamp.sec, msg->header.stamp.nsec);
```

Comment:
```suggestion
    const ros::Time msg_stamp_time(msg->header.stamp.sec, msg->header.stamp.nsec);
```
or
```suggestion
    const ros::Time msg_stamp_time{ msg->header.stamp.sec, msg->header.stamp.nsec };
```
depending on what's more common in `fuse`, which I believe it's the second case, but not sure.